code/game/turfs/open.dm

```diff
@@ -388,6 +388,10 @@
 	if (AM)
 		thing_to_check = list(AM)
 	for(var/thing in thing_to_check)
+		if(ismecha(thing))
+			var/obj/vehicle/sealed/mecha/burned_mech = thing
+			burned_mech.take_damage(rand(40, 120), BURN)
+			. = 1
 		if(isobj(thing))
```

Comment:
I would probs chuck mech at the bottom of the if chain but doesn't matter. Where it is now though, the obj if statement needs to be an else if, otherwise it's doing the mech damage as well as obj based fireact